apollo-biz/src/main/java/com/ctrip/framework/apollo/biz/service/ItemService.java

```diff
@@ -198,8 +208,22 @@ public Item update(Item item) {
     return managedItem;
   }
 
-  private boolean checkItemValueLength(long namespaceId, String value) {
+  private boolean checkItemValueLength(long namespaceId, String value, String appId, String namespaceName, String clusterName) {
```

Comment:
How about we keep the original method signature and deal the parent namespace logic within the method?

```java
  private boolean checkItemValueLength(long namespaceId, String value) {
    int limit = getItemValueLengthLimit(namespaceId);
    Namespace namespace = namespaceService.findOne(namespaceId);
    Namespace parentNamespace = namespaceService.findParentNamespace(namespace);
    if (parentNamespace != null) {
      int parentLimit = getItemValueLengthLimit(parentNamespace.getId());
      limit = Math.max(parentLimit, limit);
    }
    if (!StringUtils.isEmpty(value) && value.length() > limit) {
      throw new BadRequestException("value too long. length limit:" + limit);
    }
    return true;
  }

```